### PR TITLE
feat: add context manager to receive can message

### PR DESCRIPTION
**Pull request by yannpoupon**
Add a context manager to be able to receive a message after a send operation in case the response is too fast.

### CODE REVIEW

File: src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py [hunk 1]
```diff
@@ -130,6 +161,13 @@ def get_last_signal(self, message_name: str, signal_name: str) -> Optional[Any]:
 
         return None
 
+    def get_collected_messages(self) -> list[CanMessage]:
+        """Get all the messages collected with the context manager
+
+        :return: a list with all the collected messages
+        """
+        return deepcopy(self._messages_collected)
```
Comment (sebastianpfischer) [inline src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py:169]: You probably should clear the `self._messages_collected` list

File: src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py [hunk 2]
```diff
@@ -52,6 +78,9 @@ def __init__(self, com: CChannel, dbc_file: str, **kwargs):
         path_to_dbc_file = dbc_file
         self.parser = CanMessageParser(path_to_dbc_file)
         self.can_messages = {}
+        self._collect_msg = threading.Event()
```
Comment (sebastianpfischer) [inline src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py:81]:
```suggestion
        # Variables to manage the collection of can messages with a context-manager
        self._collect_msg = threading.Event()
```

**Review by sebastianpfischer — CHANGES_REQUESTED** [review]

File: src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py (hunk 2 above)
Comment (yannpoupon) [reply src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py:82]: Done

File: src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py (hunk 1 above)
Comment (yannpoupon) [reply src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py:170]: It is clear whenever the context manager is entered.

**Review by TedRio — APPROVED** [review]

File: src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py (hunk 1 above)
Comment (Pog3k) [reply src/pykiso/lib/auxiliaries/can_auxiliary/can_auxiliary.py:170]: I think also its nicer so get_collected_messages can be called multiple times.

**Review by Pog3k — APPROVED** [review]

**Pull request merged**